commit 6b7b54eede5c33e13fba124790c436779b8bc71f (refs/changes/32/3632/3)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-03-20T01:20:13+00:00 (1 year, 7 months ago)
    
    MANTA-3600 binder could have more trace logs
    Reviewed by: Alex Wilson <alex.wilson@joyent.com>
    Approved by: Alex Wilson <alex.wilson@joyent.com>

diff --git a/lib/zk.js b/lib/zk.js
index 5982c3b..52bef67 100644
--- a/lib/zk.js
+++ b/lib/zk.js
@@ -145,8 +145,19 @@ TreeNode.prototype.onDataChanged = function (zk, data, stat) {
         }
         this.tn_data = parsedData;
 
-        if (parsedData === null || typeof (parsedData.type) !== 'string')
+        if (parsedData === null || typeof (parsedData.type) !== 'string') {
+                this.tn_log.trace({
+                    path: this.tn_path,
+                    parsedData: parsedData,
+                    typeField: typeof (parsedData.type)
+                }, 'onDataChanged: unexpected data');
                 return;
+        }
+
+        this.tn_log.trace({
+            path: this.tn_path,
+            type: parsedData.type
+        }, 'onDataChanged');
 
         switch (parsedData.type) {
         case 'db_host':
@@ -172,6 +183,7 @@ TreeNode.prototype.onDataChanged = function (zk, data, stat) {
 };
 TreeNode.prototype.unbind = function () {
         var self = this;
+        this.tn_log.trace('unbinding node at "%s"', this.tn_path);
         if (this.tn_watcher) {
                 this.tn_watcher.removeAllListeners('childrenChanged');
                 this.tn_watcher.removeAllListeners('dataChanged');
